src/main/java/edu/kis/powp/jobs2d/drivers/monitoring/DriverMonitorDecorator.java

```diff
@@ -2,42 +2,74 @@
 
 import edu.kis.powp.jobs2d.drivers.AbstractDecorator;
 import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.monitoring.DriverEventManager.DriverEventType;
 import edu.kis.powp.observer.Publisher;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
 
 public class DriverMonitorDecorator extends AbstractDecorator {
     private final DriverUsageMonitor monitor;
     private final Publisher movePublisher = new Publisher();
-    private static boolean monitorEnabled = true;
-    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor) {
+    private final DriverMonitoringConfig config;
```

Comment:
DriverMonitoringConfig holds many dependencies. I am not sure if they are required here. e.g. DriverUsageMonitor works without any external dependencies.